src/middlewares.js

```diff
@@ -246,6 +246,11 @@ export function handleParseHeaders(req, res, next) {
       `Request using master key rejected as the request IP address '${clientIp}' is not set in Parse Server option 'masterKeyIps'.`
     );
     isMaster = false;
+    // Create a custom error with a 403 status and a specific message #BreakingChange
+    const error = new Error();
+    error.status = 403; // Forbidden status code
+    error.message = `Access denied: IP address '${clientIp}' is not authorized to use the master key.`;
+    throw error;
```

Comment:
```suggestion
    const error = new Error();
    error.status = 403;
    error.message = `unauthorized`;
    throw error;
```

I suggest to make the error message more ambiguous. We log the specific error in the server logs anyway, but an outside attacker should be getting as little info as possible.